digicampipe/image/utils.py

```diff
@@ -0,0 +1,416 @@
+from astropy.io import fits
+from astropy.stats import sigma_clipped_stats
+import numpy as np
+import scipy
+from skimage.draw import polygon
+import matplotlib.pyplot as plt
+from time import clock
+import cv2
+from scipy import signal, ndimage, optimize
+import numbers
+from photutils import DAOStarFinder
+
+
+def crop_image(image, crop_pixel1, crop_pixel2):
+    """
+    :param image: 2D array of the image data with a shape = (vertical size, horizontal size)
+    :param crop_pixel1: iterable of 2 ints: position (pos_x , pos_y) of one corner of the returned region
+    :param crop_pixel2: iterable of 2 ints: position (pos_x , pos_y) of the opposite corner of the returned region
+    :return: the cropped image, the 1st corner and the opposite corner positions
+    """
+    if type(image) is str:
+        image = fits.open(image)[0].data
+    if type(image) is not np.ndarray:
+        raise AttributeError("image must be a filename or a numpy array.")
+    if crop_pixel1 is not list and len(crop_pixel1) is not 2:
+        raise AttributeError("crop_pixel1 must be a list of 2 ints", "(position in px of one corner of the crop area).")
+    if crop_pixel2 is not list and len(crop_pixel1) is not 2:
+        raise AttributeError("crop_pixel2 must be a list of 2 ints",
+                             "(position in px of the other corner of the crop area).")
+    crop_xmin, crop_ymin = crop_pixel1
+    crop_xmax, crop_ymax = crop_pixel2
+    if not isinstance(crop_xmin, numbers.Integral) or not isinstance(crop_ymin, numbers.Integral):
+        print('Warning in crop_image(): crop_pixel1 does not contain ints')
+        crop_xmin = int(crop_xmin)
+        crop_ymin = int(crop_ymin)
```

Comment:
Often it is better to fail if a parameter has a wrong type, then to do your best to recover the mistake made by the caller. I would just state a "CropRegion" or "CropRectangle" must have integer fields: "top, bottom, left, right" or "x1, x2, y1, y2" or  whatever. If not, we do not convert to integers but simply fail. We expect the caller of this function to provide meaningful inputs and not something, which is almost good and we have to guess what the caller means. 

By simply omitting these type-checks, later this part:

    image[rectangle.bottom:rectangle.top, rectangle.left:rectangle.right]

will throw an exception if e.g. `rectangle.top` is not an integer but a float, or a string or whatever.

Reply:
fixed in 4724d7158293